### DISCUSSION TITLE
(Android) Has there been a 'Picker' regression?

### DISCUSSION THREAD

**Discussion by bradencohen** [dotnet/maui#14837] [Upgrading from Xamarin]
Please tell me if I'm doing something wrong or missing something...

On Android, the picker in Xamarin seems to be a little more enhanced than the one in MAUI:

Using a `Xamarin.Forms.Picker`:
![image](https://user-images.githubusercontent.com/11482832/235228291-d984d40e-5c2f-481f-9286-20a324dfb4d0.png)

1. Stays the same size consistently.
2. Has scrolling capabilities to select different items.

The `Microsoft.Maui.Controls.Picker`:
![image](https://user-images.githubusercontent.com/11482832/235229370-de9b80fb-f09d-44e2-97e3-47fb745c0007.png)

I purposefully duplicated the items to show how the window just expands. Both screenshots stem from a Pixel 5 - API 33.

**Comment by davidortinau** [comment]
The picker code looks to be a direct copy from Xamarin.Forms. The style change here appears to be theme/style related since we use the native Android component. MAUI updated that compared to Xamarin.Forms. 

https://developer.android.com/develop/ui/views/components/dialogs
https://m2.material.io/components/dialogs#usage
https://m3.material.io/components/dialogs/overview

The style looks accurate here, though @PureWeen mentioned we should confirm this. I do wonder about the height of the modal dialog, and the presence of the OK.

@bradencohen are there UX goals you cannot achieve in the new UI?

**Reply by bradencohen** [reply]
Gotcha. That makes sense. It seems to work fine, but I'll do some more testing (a picker with like 30 items maybe) and post it here.

**Reply by bradencohen** [reply]
The UX definitely isn't unusable: 
![image](https://user-images.githubusercontent.com/11482832/236072177-a55b8332-3172-4b26-be1f-13a1454d696d.png)

In my opinion, the experience is worse, but if that's Android's fault then so be it (what do I know anyway?). 

@davidortinau Thanks for taking a look!
